### ISSUE TITLE
Support split-dns via dnsmasq in addition to systemd-resolved

### ISSUE THREAD

**Issue by dark-penguin** [ancwrd1/snx-rs#61]
NetworkManager (i.e. most other VPNs/networks) supports split-dns via dnsmasq. Interestingly enough, you are NOT supposed to install the whole `dnsmasq` package - only `dnsmasq-base`. This mode is enabled in `/etc/NetworkManager/NetworkManager.conf`:
```
[main]
dns=dnsmasq
```
I don't think it modifies any files; it seems to communicate with dnsmasq completely in runtime. Unfortunately, this is about all the information I can provide about it - I only have knowledge about *using* it... :slightly_smiling_face: 

Could it be possible to also optionally support doing this via dnsmasq and not only systemd-resolved?

Of course, "some other project is doing this" is not a very good reason to implement stuff, but the reason here is that in order to use this VPN type "along with" other types, it would really help if they could use the same subsystem to set up DNS.

**Comment by ancwrd1** [comment]
I think I tried it already using nmcli to set DNS servers but it didn't work for some reason. Didn't look too deep though.

**Comment by dark-penguin** [comment]
Looks like this is not a very good idea in general. NetworkManager launches its own instance of dnsmasq, so others are not supposed to touch it.

Apparently systemd-resolved is "the modern way" to do it, and NetworkManager supports it too - it's just that I'm not modern enough to learn it. I'll see if I can migrate to systemd-resolved instead.

But snx-rs will properly add resolvers only for each forward zone, and only for each reverse zone, correctly determining reverse zones for each server-provided subnets like `172.20.0.0/23`, right? (Adding `0.20.172.in-addr.arpa` would correspond to `172.20.0.0/24`, so there are additional records required to correctly map only specified subnets.)

**Comment by dark-penguin** [comment]
Is it possible to see what zones did the server push? I'm running the service with "info" log level, but I don't see this information in journald.

**Comment by ancwrd1** [comment]
You can see it in debug mode.
`systemd-resolved` migration is simple, it's just a symlink and NM recognizes it automatically. I have described it in the README file.

**Comment by dark-penguin** [comment]
Yes, I saw it, but I did not know that NetworkManager also supports it - and that it works even better than dnsmasq nowadays. the situation was very different the last time I checked. :smile: 

I would need to study a new solution before migrating to it! :slightly_smiling_face: How does it function, how to check what is it doing, how to diagnose it.

Figuring that out was indeed pretty straightforward! And everything seems to work just fine right after the migration. Makes sense that no one uses dnsmasq anymore.

I suggest we close this issue - taking the effort to support dnsmasq doesn't seem to make much sense given the situation.
